gnomad/utils/filtering.py

```diff
@@ -460,12 +465,39 @@ def filter_to_gencode_cds(
         "This Gencode CDS interval filter does not filter by transcript! Please see the"
         " documentation for more details to confirm it's being used as intended."
     )
-    filter_expr = hl.is_defined(gencode_ht[t.locus])
+    if genes:
+        genes_upper = (
+            [genes.upper()] if isinstance(genes, str) else [g.upper() for g in genes]
+        )
+        gencode_ht = gencode_ht.filter(
+            hl.literal(genes_upper).contains(gencode_ht.gene_name)
+        )
 
-    if isinstance(t, hl.MatrixTable):
-        t = t.filter_rows(filter_expr)
+    if padding:
+        gencode_ht = gencode_ht.annotate(
+            padded_interval=hl.locus_interval(
+                gencode_ht.interval.start.contig,
+                gencode_ht.interval.start.position - padding,
+                gencode_ht.interval.end.position + padding,
+                includes_start=gencode_ht.interval.includes_start,
+                includes_end=gencode_ht.interval.includes_end,
+                reference_genome=gencode_ht.interval.start.dtype.reference_genome,
+            )
+        )
+        gencode_ht = gencode_ht.key_by("padded_interval")
+
+    if genes or padding:
```

Comment:
I'm not sure about this. I think even if padding is true and genes is false, it will have that issue. What if instead we have a parameter that specifies the number of intervals we are willing to collect and set the default to something that filter_intervals would still be the better option

Reply:
I think this is a good idea, TTN seems to have the most exons, 363, how about 1000 intervals? Would that cause memory issue based on your experience?

Reply:
I'm not sure, I haven't ever really tried

Reply:
I set it to 3000, based on the distribution of CDS per gene and the runtime in my notebook: 
![Screenshot 2024-12-20 at 2 54 29 PM](https://github.com/user-attachments/assets/3a65d698-4900-4ffc-825d-5e32dbfbcb60)

![Screenshot 2024-12-20 at 2 51 19 PM](https://github.com/user-attachments/assets/58d47e47-8a65-43f5-8652-9559c47aeb4b)
lmk what you think.

Reply:
2 minutes is not so bad?

Reply:
works for me